lib/publiccloud/basetest.pm

```diff
@@ -152,6 +153,16 @@ sub _cleanup {
 
 sub post_fail_hook {
     my ($self) = @_;
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        sles4sap_cleanup(
```

Comment:
what about moving 
```
   # If there's an open ssh connection to the VMs, return to host console first
    select_host_console(force => 1);
    ``` 
    before line 157?

Reply:
> You mean move it to the PC basetest module? But then, it wouldn't run in the case a module that inherits our own basetest class runs between the ssh start/end modules

Are there any sles4sap test modules scheduled in between `ssh_interactive_start` and `_end` ?

Reply:
My thought is that `sles4sap_cleanup` could be more "generic" if it does not call `select_host_console`.
Maybe has not to be: maybe one of the duty of this function has to be "This function has a set of commands that are mind to be executed from the JumpHost and not from any cloud nodes, be sure to be executed like this".

Reply:
> Are there any sles4sap test modules scheduled in between ssh_interactive_start and _end ?

there's mr_test, but it inherits a completely different parent (sles4sap). 

> My thought is that sles4sap_cleanup could be more "generic"

Is there a chance we'll ever need to run cleanup on a cloud VM? Otherwise, we can still use it outside of a `post_fail_hook` - the console command would not have any effect in that case. But I can move it, if you think it's better